controllers/nodefeaturediscovery_controller.go

```diff
@@ -162,17 +162,115 @@ func (r *NodeFeatureDiscoveryReconciler) Reconcile(ctx context.Context, req ctrl
 
 	r.Log.Info("Ready to apply components")
 	nfd.init(r, instance)
+	result, err := applyComponents()
+	if err != nil {
+		return ctrl.Result{Requeue: true}, err
+	}
+
+	// Check the status of the NFD Operator Worker ServiceAccount
+	if rstatus, err := r.getWorkerServiceAccountConditions(ctx); err != nil {
+		return r.updateDegradedCondition(instance, conditionFailedGettingNFDWorkerServiceAccount, err)
+	} else if rstatus.isDegraded {
+		return r.updateDegradedCondition(instance, err.Error(), err)
```

Comment:
Hmm, looks still very suspicious, shouldn't the condition reason to be something else than "nil" 🧐  Also, the message should be a human readable description, right(?)

```suggestion
		return r.updateDegradedCondition(instance, conditionNFDWorkerServiceAccountDegraded, "nfd-worker service account has been degraded")
```

...and ditto X times below

Reply:
will do for all of them